### PR TITLE
Disallow syntax rules that match the empty string (for now)

### PR DESCRIPTION
**Pull request by maxbrunsfeld**
These *epsilon rules* are actually somewhat complicated to handle; they've never worked 100% correctly. For now, I'm just going to disallow them.

### CODE REVIEW

**Comment by joshvera** [comment]
👍 do you think this might affect any of our parsers?

**Comment by maxbrunsfeld** [comment]
Just ruby, but it's a trivial adjustment to make it work.

File: src/compiler/build_tables/build_parse_table.cc [hunk 1]
```diff
@@ -177,7 +177,7 @@ class ParseTableBuilder {
             parse_table.add_terminal_action(state_id, lookahead, action);
           } else {
             ParseAction &existing_action = entry.actions[0];
-            if (allow_any_conflict) {
+            if (existing_action.type == ParseActionTypeAccept || allow_any_conflict) {
```
Comment (joshvera) [inline src/compiler/build_tables/build_parse_table.cc]: What does this change have to do with `TSCompileErrorTypeEpsilonRule` or the `flatten_grammar` function?

Comment (maxbrunsfeld) [inline src/compiler/build_tables/build_parse_table.cc]: Pretty much nothing; it's fixing an unrelated bug. But the integration test that I wrote for epsilon rules crashed because it triggered this edge case, so I'm just including this as a drive-by fix.

Comment (joshvera) [inline src/compiler/build_tables/build_parse_table.cc]: 👌
